src/bootstrap/builder.rs

```diff
@@ -455,6 +455,7 @@ impl<'a> Builder<'a> {
                 test::CompiletestTest,
                 test::RustdocJSStd,
                 test::RustdocJSNotStd,
+                test::RustdocJSParser,
```

Comment:
It seems unfortunate to keep adding more custom test suites. Could we instead do run-make tests for these? Or if that's too much of a pain, condense all the `node` tests into one somehow?

cc @Mark-Simulacrum

Reply:
Would it be possible to add a way, in the existing rustdoc-js test system, to have the `EXPECTED` work the way it does now, and have `PARSED` for the JSON parse tree?

For example, this would be a test case. It combines the parser check with the search check, and would go in the same test suite.

```js
const QUERY = ['"R"<"P">'];

 const PARSED = [
     {
         args: [],
         elemName: null,
         elems: [{
             isExact: true,
             name: "r",
             fullPath: ["r"],
             pathWithoutLast: [],
             pathLast: "r",
             generics: [
                 {
                     isExact: true,
                     name: "p",
                     fullPath: ["p"],
                     pathWithoutLast: [],
                     pathLast: "p",
                     generics: [],
                 },
             ],
         }],
         foundElems: 1,
         id: "\"R\"<\"P\">",
         nameSplit: null,
         original: "\"R\"<\"P\">",
         returned: [],
         typeFilter: -1,
         val: "\"r\"<\"p\">",
     }
 ];

const EXPECTED = [
     {
         'returned': [
             { 'path': 'generics', 'name': 'alef' },
         ],
        'in_args': [
            { 'path': 'generics', 'name': 'alpha' },
         ],
     }
];
```

Reply:
I guess we could make one of the two optional for parser only checks. Sounds like a great idea!